ruby/red-arrow/lib/arrow/array-builder.rb

```diff
@@ -121,15 +126,28 @@ def detect_builder_info(value, builder_info)
             detected: true,
           }
         when BigDecimal
-          if value.to_arrow.is_a?(Decimal128)
-            {
-              builder: Decimal128ArrayBuilder.new,
-            }
-          else
+          builder_info ||= {}
+          if builder_info[:builder]
             {
-              builder: Decimal256ArrayBuilder.new,
+              builder: StringArrayBuilder.new,
               detected: true,
             }
+          else
+            precision = [builder_info[:precision] || 0, value.precision].max
+            scale = [builder_info[:scale] || 0, value.scale].max
```

Comment:
These 2 lines don't permit overwriting the value's precision and the scale with smaller values by specifying the precision and the scale in `builder_info`.  Is it intentional?

Reply:
Could you provide an example case?
`Arrow::Array.new([BigDecimal("1.1"), BigDecimal("11.11")])`?

Reply:
I misreading the code.  It's OK now.  But, I guess we may need to permit to pass precision and scale to `Arrow::Array.new` as optional arguments like`Arrow::Array.new([BigDecimal("1.1"), BigDecimal("11.11")], scale: 1)`.

Reply:
If users know an expected data type, they should use `Arrow::XXXArray.new` instead of `Arrow::Array.new` such as `Arrow::Decimal128Array.new({scale: ..., precision: ...}, [BigDecimal(...). ...])`.

`Arrow::Array.new` may not build a `Arrow::Decimal128Array`. In the case, the given optional arguments are just ignored. It may confuse users.

Reply:
I see.